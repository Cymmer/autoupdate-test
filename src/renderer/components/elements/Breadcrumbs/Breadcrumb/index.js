import React from 'react';
import { Link } from 'react-router-dom';
import cn from 'classnames';
import PropTypes from 'prop-types';
import styles from './styles.module.scss';

import Icon from '../../Icon';

const Breadcrumb = ({ children, link, active }) => (
  <Link
    className={cn({
      [styles.Breadcrumb]: !active,
      [styles.Breadcrumb___active]: active,
    })}
    to={link}
  >
    <span className={styles.Breadcrumb_text}>{children}</span>
    {!active && (
      <Icon
        className={styles.Breadcrumb_icon}
        icon="chevron_right"
        id="chevronRight"
      />
    )}
  </Link>
);

Breadcrumb.defaultProps = {
  active: false,
};

Breadcrumb.propTypes = {
  children: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
  active: PropTypes.bool,
};

export default Breadcrumb;
