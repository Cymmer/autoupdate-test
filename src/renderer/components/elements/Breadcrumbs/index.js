import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import styles from './styles.module.scss';

import Breadcrumb from './Breadcrumb';

const Breadcrumbs = ({ className, links }) => (
  <div className={cn(styles.Breadcrumbs, className)}>
    {links &&
      links.map((link, index) => (
        <Breadcrumb
          key={link.name}
          active={index === links.length - 1}
          link={link.link}
        >
          {link.name}
        </Breadcrumb>
      ))}
  </div>
);

Breadcrumbs.propTypes = {
  className: PropTypes.string,
  links: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      link: PropTypes.string,
    })
  ),
};

export default Breadcrumbs;
