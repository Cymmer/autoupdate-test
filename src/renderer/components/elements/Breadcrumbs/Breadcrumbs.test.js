import React from 'react';
import { shallow } from 'enzyme';
import Breadcrumbs from './index';

describe('RENDERING - Breadcrumbs', () => {
  it('should render with default props', () => {
    const component = shallow(<Breadcrumbs />);
    expect(component).toMatchSnapshot();
  });

  it('should render appropriate number of links', () => {
    const component = shallow(
      <Breadcrumbs
        links={[
          {
            name: 'Link 1',
            link: 'www.codechum.com',
          },
          {
            name: 'Link 2',
            link: 'app.codechum.com',
          },
        ]}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
