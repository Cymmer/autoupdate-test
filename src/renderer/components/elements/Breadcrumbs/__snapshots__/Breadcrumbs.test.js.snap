// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Breadcrumbs should render appropriate number of links 1`] = `
<div
  className="Breadcrumbs"
>
  <Breadcrumb
    active={false}
    key="Link 1"
    link="www.codechum.com"
  >
    Link 1
  </Breadcrumb>
  <Breadcrumb
    active={true}
    key="Link 2"
    link="app.codechum.com"
  >
    Link 2
  </Breadcrumb>
</div>
`;

exports[`RENDERING - Breadcrumbs should render with default props 1`] = `
<div
  className="Breadcrumbs"
/>
`;
